### ISSUE TITLE
[Bug]: Fluid Text Area - enable counter prop broken style

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14035]
### Package

@carbon/react

### Package version

carbon/react@1.31.0

### Description

Currently in Experimental Fluid TextArea if you enable the `enableCounter` prop the counter overlaps with the label 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/experimental-unstable-fluidtextarea--playground&args=enableCounter:true

### Steps to reproduce

https://github.com/carbon-design-system/carbon/assets/32720851/e25015d9-f588-4cf4-8563-b1e17b08b1cc

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
It would be great if we could cover this with VRT as part of fixing this issue. We may be able to update `snapshotStory` to take in some URL query parameters or something to enable the control.

**Comment by guidari** [comment]
About VRT part. @tay1orjones 

I tried to add on `snapshotStory` the id as `experimental-unstable-fluidtextarea--playground&args=enableCounter:true` to check if would work, but it doesn't. 

I did also research about topic but couldn't get much information on how to do it.

One question I've been thinking is about **billing**. That would open a spot for more VRT components checks like that, which would increase the `snapshotStory` and, by consequence, the billing on Percy (I suppose, I'm not familiar with the billing). 

I'm happy to open an issue if that makes sense, so we could tackle the details more in depth.
